opi/__init__.py

```diff
@@ -227,13 +221,13 @@ def get_keys_from_rpmdb():
 		})
 	return keys
 
-def install_packages(packages, from_repo=None, allow_vendor_change=False, allow_arch_change=False, allow_downgrade=False, allow_name_change=False):
+def install_packages(packages, action='in', from_repo=None, allow_vendor_change=False, allow_arch_change=False, allow_downgrade=False, allow_name_change=False):
```

Comment:
I would prefer to not have a function `install_packages` that does something different specified via an argument.
But let's change the code like this (untested):

```patch
diff --git a/opi/__init__.py b/opi/__init__.py
index abf23cc..662a837 100644
--- a/opi/__init__.py
+++ b/opi/__init__.py
@@ -99,7 +99,7 @@ def add_packman_repo(dup=False):
 	)
 
 	if dup:
-		install_packman_packages([], action='dup', allow_downgrade=True, allow_vendor_change=True)
+		dist_upgrade(allow_downgrade=True, allow_vendor_change=True)
 
 def add_openh264_repo(dup=False):
 	project = get_os_release()['NAME']
@@ -123,7 +123,7 @@ def add_openh264_repo(dup=False):
 		)
 
 	if dup:
-		install_packages([], action='dup', from_repo=repo, allow_downgrade=True, allow_vendor_change=True)
+		dist_upgrade(from_repo=repo, allow_downgrade=True, allow_vendor_change=True)
 
 def install_packman_packages(packages, **kwargs):
 	install_packages(packages, from_repo='packman', **kwargs)
@@ -264,7 +264,13 @@ def get_keys_from_rpmdb():
 		})
 	return keys
 
-def install_packages(packages, action='in', from_repo=None, allow_vendor_change=False, allow_arch_change=False, allow_downgrade=False, allow_name_change=False):
+def install_packages(packages, **kwargs):
+	return pkgmgr_action('in', packages, **kwargs)
+
+def dist_upgrade(**kwargs):
+	return pkgmgr_action('dup', **kwargs)
+
+def pkgmgr_action(action, packages=[], from_repo=None, allow_vendor_change=False, allow_arch_change=False, allow_downgrade=False, allow_name_change=False):
 	if get_backend() == BackendConstants.zypp:
 		args = ['sudo', 'zypper', action]
 		if from_repo:
```

Reply:
LGTM